lib/ecstatic.js

```diff
@@ -330,6 +339,19 @@ var ecstatic = module.exports = function (dir, options) {
 ecstatic.version = version;
 ecstatic.showDir = showDir;
 
+function hasGzipId12(gzipped, cb) {
+  var stream = fs.createReadStream(gzipped, { start: 0, end: 1, autoClose: true }),
+      isGzip = false;
+
+  stream.on("close", function () {
+    cb(isGzip);
+  });
+
+  stream.on("data", function (chunk) {
```

Comment:
We can talk about removing 0.12 support, given that using new Buffer is apparently deprecated because node core likes breaking shit.